include/eve/detail/function/simd/riscv/subscript.hpp

```diff
@@ -0,0 +1,112 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/arch/riscv/rvv_common_masks.hpp>
+
+namespace eve::detail
+{
+template<typename T, typename N>
+EVE_FORCEINLINE T
+extract(wide<T, N> const& v, std::size_t i) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  auto OnFirstNeeded = __riscv_vslidedown_tu(v, v, i, N::value);
+  if constexpr( std::is_floating_point_v<T> ) return __riscv_vfmv_f(OnFirstNeeded);
+  else return __riscv_vmv_x(OnFirstNeeded);
+}
+
+template<typename T, typename N>
+EVE_FORCEINLINE void
+insert(wide<T, N>& v, std::size_t i, std::convertible_to<T> auto x) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  // get mask with 1 on i'th element.
+  auto mask = rvv_one<T, N>(i);
+  v         = if_else(mask, static_cast<T>(x), v);
+}
+
+template<typename T, typename N>
+EVE_FORCEINLINE logical<T>
+                extract(logical<wide<T, N>> const                &v, std::size_t i) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  using u_m1_cardinal               = typename riscv_rvv_dyn_::m1_fixed<unsigned>;
+  using inter_type                  = wide<unsigned, u_m1_cardinal>;
+  auto             uint_wide        = bit_cast(v, as<inter_type> {});
```

Comment:
I don't get this. Why are you bitcasting specifically to uint32_t

Reply:
This is comes due to restrictions of RISC-V vector ISA. For storing logical, I use rvv intrinsic type called mask. This is in fact bits, where n-th bit refer to the value of logical in n-th position. Hovewer, in RISC-V ISA I do not know any way to get n-th bit from mask, so I cast it to several unsigned, get interesting unsigned, and in it find the needed bit.

Let me show you example. Note: for simplification, I will use bellow casting to `unsigned char`, but in terms of effectiveness `unsigned` will be better. Suppose you have `logical<wide<int, 16>>=={1,0,1,0,0,1,0,1,0,1,0,1,1,01,0}`.  It will be stored in RISC-V vector register as:
`x....x1010010101011010`<sub>2</sub>, where `x` means that value is unknown. To get, for example, 10th bit, I do the following:
1) bitcast this vector to wide type (`unsigned char` is used for convince), that will occupy one vector register (note: there can not be used part of vector register, as for vector mask we occupy one full register). By the way, that is why I need `m1_fixed` - to get wide type that could be used to occupy only one vector register.
In result, for vlen=128, I will have `wide<unsigned char, 16>` with values:
<...`x`<sub>2</sub>,`x`<sub>2</sub>,`10100101`<sub>2</sub>,`01011010`<sub>2</sub>>
2) Get from casted `wide` scalar, where we have interesting for us bit (we still have no single instruction, but there we have some ways to do this), so we have `10100101`<sub>2</sub>
3) Get from resulted scalar interesting for us bit. There it will be I suppose `1`.

Also bellow I do the same thing when I need to insert bool into logical.

Reply:
This function seems very questionable. Just store the mask to the stack and return a bit from the array. We can think of something better later.

Reply:
Done. Though I suppose later I will create a separate review with this optimization.